pulsar-client-cpp/lib/auth/athenz/ZTSClient.cc

```diff
@@ -143,14 +143,17 @@ std::string ZTSClient::ybase64Encode(const unsigned char *input, int length) {
 char *ZTSClient::base64Decode(const char *input) {
     BIO *bio, *b64;
     size_t length = strlen(input);
-    char *result = (char *)malloc(length);
+    char *result = (char *)calloc(length, sizeof(char));
 
     bio = BIO_new_mem_buf((void *)input, -1);
     b64 = BIO_new(BIO_f_base64());
     bio = BIO_push(b64, bio);
 
     BIO_set_flags(bio, BIO_FLAGS_BASE64_NO_NL);
-    BIO_read(bio, result, length);
+    int decodeStrLen = BIO_read(bio, result, length);
+    if(decodeStrLen > 0) {
+        result[decodeStrLen] = '\0';
+    }
```

Comment:
We have a couple of options here (error cases).
1) Call free(result) and return nullptr -- caller should check if the returned value is not nullptr
2) Put null at result[0] -- caller should check if the length of returned string is not zero, and then free it

Without 1) or 2), callers cannot avoid access violation. Of course we can avoid it by using calloc() too, but that's basically the same as 2) except efficiency.

Reply:
I implemented it in 2).
Please check,again.